### PR TITLE
Refactor editor containers and their usage in MainFrame

### PR DESCRIPTION
**Pull request by 4ian**
* Ensure all editor are properly flow typed - hopefully avoiding any regression in the future.
* Avoid using potentially-stale closures in MainFrame for editors that would be bug prone.
* Slightly reduce the complexity inside MainFrame now that these risky closures are not necessary anymore. Make the file a bit shorter :)

### CODE REVIEW

**Comment by HarsimranVirk** [comment]
@4ian is the MainFrame stable enough now? If it is then maybe I could start working on the Open Recent Project PR 😅

File: newIDE/app/src/MainFrame/index.js [hunk 1]
```diff
@@ -211,6 +213,7 @@ const MainFrame = (props: Props) => {
     openSubscriptionDialog,
   ] = React.useState<boolean>(false);
   const [exportDialogOpen, openExportDialog] = React.useState<boolean>(false);
+  const preferences = React.useContext(PreferencesContext);
 
   React.useEffect(() => {
```
Comment (HarsimranVirk) [inline newIDE/app/src/MainFrame/index.js:218]: Reminder: This has to be removed if it isn't needed anymore :)

Comment (4ian) [reply newIDE/app/src/MainFrame/index.js:218]: Yep! I'll remove this soon if we don't find any issues.

**Comment by 4ian** [comment]
> is the MainFrame stable enough now? If it is then maybe I could start working on the Open Recent Project PR 😅

I think it is! At least if there is an issue, I feel confident fixing it quickly :)

File: newIDE/app/src/MainFrame/index.js (hunk 1 above)
Comment (4ian) [reply newIDE/app/src/MainFrame/index.js:218]: Thanks for the reminder ;)

**Pull request merged**